### ISSUE TITLE
Better Support for Troubleshooting

### ISSUE THREAD

**Issue by dcjulian29** [thinktecture/Thinktecture.IdentityServer.v3.Documentation#32]
Please add support for configuring verbose and detailed error messages.

**TLDR;**

I've used NuGet to install the IdentityServer into an MVC site based on the Getting Started documentation. I've got the server authenticating correctly on my local system using a self-signed certificate. I then moved over to my "QA" server (2012 R2/IIS 8.5) which is hosted at another location but all I can get after logging in is "There was an unexpected error" every time I attempt to log in. The only difference between the two sites is the Signing Certificate. On localhost, I load a self-signed "CN=localhost" pfx file. On the QA server, which uses a wild-card SSL certificate, I load the certificate from the "Web Hosting" certificate store. I've validated that I'm getting the certificate by creating an unauthenticated view to show the Certificate Subject Name and Thumbprint. Unfortunately, other than deploying code and the web browser, I have no access to the hosting server and the IIS AppPool identity doesn't have "write" permissions to the file system due to corporate policies.

I've search the web and followed the documentation to enable tracing for the application. Either the document is out-of-date or is not correct as no tracing information is output even when I'm running it on my local system. I validated that my tracing configuration was enabled by writing tracing information during the "startup.cs" class...

If I can not resolve the "generic" log in error soon, I'll have to abandon (IMHO) a superior authentication/authorization solution and return to an internal (most likely insecure) authentication solution... My hope by creating this issue is that some attention is given to when developers have to troubleshoot errors in this code on servers that they do not have direct access to...

**Comment by brockallen** [comment]
Closing this since there's already another active thread for your question.
